bundles/io/org.openhab.io.transport.mqtt/src/main/java/org/openhab/io/transport/mqtt/MqttService.java

```diff
@@ -91,7 +91,13 @@ public void updated(Dictionary<String, ?> properties) throws ConfigurationExcept
             } else if (property.equals("async")) {
                 conn.setAsync(Boolean.parseBoolean(value));
             } else if (property.equals("clientId")) {
+                if (value.length() > 65535) {
+                    logger.warn("clientId must be less than 65536 characters long");
+                    throw new ConfigurationException("clientId", "clientId must be less than 65536 characters long");
+                }
                 conn.setClientId(value);
+            } else if (property.equals("allowLongerClientIds")) {
+                conn.setAllowLongerClientIds(Boolean.parseBoolean(value));
```

Comment:
I have now tried various orderings of the keys in openhab.cfg, and I have yet to find any inconsistent results. No matter whether the `allowLongerClientIds` key is specified before or after the `clientId` key, we still get the proper behaviour when the broker starts up:

21:59:28.001 [INFO ] [o.i.t.m.i.MqttBrokerConnection:112  ] - Starting MQTT broker connection 'ruby'
21:59:28.003 [DEBUG] [o.i.t.m.i.MqttBrokerConnection:322  ] - Creating new client for 'tcp://io.adafruit.com:1883' using id 'ThisIsAReallyLongClientId' and file store 'C:\Users\me\AppData\Local\Temp\/ruby'

Does this fully address your concern?